### DISCUSSION TITLE
Issue with  Positioning Light Source outside the model cuboid

### DISCUSSION THREAD

**Discussion by Anita960** [ankrh/MCmatlab#27] [General]
Dear Anders and Dominik,

I'm using MCmatlab to simulate light-tissue interactions and am encountering an issue with positioning the light source. Specifically, I want to place the light source at different distances above the XY plane (Z=0) and compare the results. For example, I am trying to compare the absorption rate with the light source at 40 cm and 20 cm above the tissue.

In my setup, I set the focus z component to -0.025 cm and observed differences in absorption rates compared to when the focus is at z=0. I set the absorption coefficient to an extremely high value so that the light source distribution would be visible. Upon reviewing the code in MCmatlablib.c, I found that results are only documented when the photon is inside the medium (G).

If results are only documented within the medium, why do I see differences in the light source distribution? (the first one is when z = -0.015, the second one is )
 Am I misunderstanding the meaning of focus? 
Can I set the z component of focus to a negative value to simulate the light source being outside the medium (G)? If not, is there a way to adjust the number or position of light sources (especially outside the defined skin layer) using the current functions?

Best regards,
Anita
![z = 0cm ](https://github.com/user-attachments/assets/27422797-97ee-4492-986a-7a4bc8afab7f)
![z= -0 025cm](https://github.com/user-attachments/assets/a41bf9f4-eb28-4fe4-aa52-63a014e5dbe4)

**Comment by ankrh** [comment]
I believe your results are correct and you are seeing the correct distribution of light for a light source positioned either exactly on top of the cuboid or some distance above the cuboid. You may be misunderstanding what happens and what does not happen outside the cuboid.

Your light source is a disk of radius 0.01 cm with an angular spread of pi/3, so it's something similar in nature to a very small circular LED or a highly divergent non-Gaussian focused laser beam.

When MCmatlab launches photons, it calculates "when a photon is emitted from the light source with a randomly generated xy position and a randomly generated theta phi angle, where will this ray intersect the z=0 slice?" and then it creates the ray at that location and starts to propagate it down into the medium. That's why the distribution is wider when you put zfocus = -0.025 cm. The source is higher up so the light is more spread out when it hits the cuboid.

If your boundarytype is 1 (all boundaries are escaping), then as soon as a ray hits any of the boundaries in the outgoing direction, it is no longer simulated and simply "escapes" from the medium. With the other boundarytypes, photons can travel outside the cuboid and will experience scattering and absorption, but the fluence rate / absorption data is only kept track of for the volume within the cuboid. But I don't think this has anything to do with your question.

**Comment by Anita960** [comment]
Dear Anders and Dominik,

Thank you for your detailed explanation; it has clarified my doubts. However, I have some questions regarding the AID.width while working on light source simulations over the past few weeks.

From my understanding, AID.width should be represented as shown in Figure 1. the normalized Absorption on the yz plane (x=0， 
The labels on the image are incorrect; the x-axis represents the y values, and the y-axis represents the z values.), when the light intensity reaches 
1/e^2 of its maximum value, the angle is the connection line between the highest point of the light intensity and the point where it reaches 1/e^2 of its maximum value, just like the yellow line shown in the figure.

While I was trying to analyze the light intensity data stored in the NAslice from the yz slice, I found that the points where the light intensity reaches 1/e^2are mostly located along the vertical line from the highest intensity point (indicated by the red dashed line in the figure).

I would like to ask if there might be a misunderstanding on my part regarding AID.width, or more specific if AID.width corresponds to a specific parameter in the Gaussian distribution after I have the light source's Gaussian profile.


I look forward to your reply!
Thank you for your help!
Best regards


Here are the code for MCsimulation and for absorption

```
%% Geometry definition
MCmatlab.closeMCmatlabFigures();
model = MCmatlab.model;

model.G.nx                = 101; % Number of bins in the x direction
model.G.ny                = 101; % Number of bins in the y direction
model.G.nz                = 150; % Number of bins in the z direction
model.G.Lx                = .1; % [cm] x size of simulation cuboid
model.G.Ly                = .1; % [cm] y size of simulation cuboid
model.G.Lz                = .1; % [cm] z size of simulation cuboid

model.G.mediaPropertiesFunc = @mediaPropertiesFunc; % Media properties defined as a function at the end of this file
model.G.geomFunc          = @geometryDefinition; % Function to use for defining the distribution of media in the cuboid. Defined at the end of this m file.

model = plot(model,'G');

%% Monte Carlo simulation
model.MC.simulationTimeRequested  = .1; % [min] Time duration of the simulation
model.MC.matchedInterfaces        = true; % Assumes all refractive indices are the same
model.MC.boundaryType             = 1; % 0: No escaping boundaries, 1: All cuboid boundaries are escaping, 2: Top cuboid boundary only is escaping, 3: Top and bottom boundaries are escaping, while the side boundaries are cyclic
model.MC.wavelength               = 532; % [nm] Excitation wavelength, used for determination of optical properties for excitation light

model.MC.lightSource.sourceType   = 4; % 0: Pencil beam, 1: Isotropically emitting line or point source, 2: Infinite plane wave, 3: Laguerre-Gaussian LG01 beam, 4: Radial-factorizable beam (e.g., a Gaussian beam), 5: X/Y factorizable beam (e.g., a rectangular LED emitter)

% For a pencil beam, the "focus" is just a point that the beam goes
% through, here set to be the center of the cuboid:
model.MC.lightSource.xFocus       = 0; % [cm] x position of focus
model.MC.lightSource.yFocus       = 0; % [cm] y position of focus
model.MC.lightSource.zFocus       = 0; % [cm] z position of focus
model.MC.lightSource.focalPlaneIntensityDistribution.radialDistr = 1; % Radial focal plane intensity distribution - 0: Top-hat, 1: Gaussian, Array: Custom. Doesn't need to be normalized.
model.MC.lightSource.focalPlaneIntensityDistribution.radialWidth = 0; % [cm] Radial focal plane 1/e^2 radius if top-hat or Gaussian or half-width of the full distribution if custom
D = 30;
R = deg2rad(D);
model.MC.lightSource.angularIntensityDistribution.radialDistr = 1; % Radial angular intensity distribution - 0: Top-hat, 1: Gaussian, 2: Cosine (Lambertian), Array: Custom. Doesn't need to be normalized.

model.MC.lightSource.angularIntensityDistribution.radialWidth = R; % [rad] Radial angular 1/e^2 half-angle if top-hat or Gaussian or half-angle of the full distribution if custom. For a diffraction limited Gaussian beam, this should be set to model.MC.wavelength*1e-9/(pi*model.MC.lightSource.focalPlaneIntensityDistribution.radialWidth*1e-2))



%model.MC.lightSouce.phi= R;

% These lines will run the Monte Carlo simulation with the provided
% parameters and subsequently plot the results:
model = runMonteCarlo(model);
model = plot(model,'MC');

%% Geometry function(s) (see readme for details)
% A geometry function takes as input X,Y,Z matrices as returned by the
% "ndgrid" MATLAB function as well as any parameters the user may have
% provided in the definition of Ginput. It returns the media matrix M,
% containing numerical values indicating the media type (as defined in
% mediaPropertiesFunc) at each voxel location.
%% Geometry function(s) (see readme for details)
function M = geometryDefinition(X,Y,Z,parameters)
M = ones(size(X)); % Air
end
%% Media Properties function (see readme for details)
% The media properties function defines all the optical and thermal
% properties of the media involved by filling out and returning a
% "mediaProperties" array of "mediumProperties" objects with various
% properties. The j indices are the numbers that are referred to in the
% geometry function (in this case, 1 for "air" and 2 for "standard tissue")
% See the readme file or the examples for a list of properties you may
% specify. Most properties may be specified as a numeric constant or as
% function handles.
% 
% The function must take one input; the cell array containing any special
% parameters you might specify above in the model file, for example
% parameters that you might loop over in a for loop. In most simulations
% this "parameters" cell array is empty. Dependence on wavelength is shown
% in examples 4 and 23. Dependence on excitation fluence rate FR,
% temperature T or fractional heat damage FD can be specified as in
% examples 12-15.
function mediaProperties = mediaPropertiesFunc(parameters)
% Always leave the following line in place to initialize the
% mediaProperties array:
mediaProperties = MCmatlab.mediumProperties;

% Put in your own media property definitions below:
j=1;
mediaProperties(j).name  = 'air';
mediaProperties(j).mua   = 1e-8; % Absorption coefficient [cm^-1]
mediaProperties(j).mus   = 1e-8; % Scattering coefficient [cm^-1]
mediaProperties(j).g     = 1; % Henyey-Greenstein scattering anisotropy

end



Here is the code for validation
```
```
NAslice = squeeze(model.MC.NA(51,:,:));
figure(100);
imagesc(model.G.y,model.G.z,NAslice.');
hold on;
%col = colorbar('XTickLabel', {'10^{2}','10^{3}','10^{4}', '10^{5}', '10^{6}'}, 'XTick',log(10.^(2:6))/log(10));
%caxis([4 6]);
colorbar;
colormap(turbo);
hold on;

xlabel('Y');
ylabel('Z');
title('x=0,yz plane');

tolerance = 0.0001;
[max_value, max_idx] = max(NAslice(:));
[y_max, z_max] = ind2sub(size(NAslice), max_idx);
target_value = max_value / exp(2);


[row_idx, col_idx] = find(abs(NAslice - target_value) < tolerance); 点
fprintf('maximun: %.4f\n', max_value);
fprintf('position of maximun: y = %d, z = %d\n', y_max, z_max);


fprintf('target point position:\n');
for i = 1:length(row_idx)
fprintf('y = %d, z = %d\n', row_idx(i), col_idx(i));
end


figure(100);
imagesc(model.G.y, model.G.z, NAslice.');
hold on;

for i = 1:length(row_idx)
% Get the corresponding y and z values
y_value = model.G.y(row_idx(i)); % Get the y value using row_idx
z_value = model.G.z(col_idx(i)); % Get the z value using col_idx

% Print the found coordinates
fprintf('point: y = %.4f, z = %.4f\n', y_value, z_value);
    
% Plot these points on the figure
 plot(y_value, z_value, 'r*', 'MarkerSize', 8); 

 end

hold off;

% Set axis properties
 xlabel('Y');
 ylabel('Z');
 title('x=0, yz plane with Target Points');
 colorbar;
 colormap(turbo);
```



![figure 1](https://github.com/user-attachments/assets/38e5fa4b-ef9b-46a8-a98b-ddc0afd6b6e6)
![angle30validation](https://github.com/user-attachments/assets/620dc4e0-5261-4201-be15-b3daee78ff02)

**Comment by ankrh** [comment]
Hi again Anita

The "maximum" value you should use for the 1/e2 consideration should not be the (0,0,0) point, it should be the maximum within the xy slice you're considering. So consider instead the xy slice at the bottom of the cuboid NA(:,:,end) and do the same 1/e2 calculation there. Then it should fit with the specified AID.width angle.
